eva-lib/src/test/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryTest.java

```diff
@@ -272,37 +287,163 @@ public void testFindByRegionsAndComplexFilters() {
 
         regions.add(new Region("11", 180205, 180221)); //2
 
         testFindByRegionsAndComplexFiltersHelper(regions, null, null, VariantEntityRepository.RelationalOperator.NONE,
                                                  null, VariantEntityRepository.RelationalOperator.NONE, null,
-                                                 VariantEntityRepository.RelationalOperator.NONE, null, 11);
+                                                 VariantEntityRepository.RelationalOperator.NONE, null, null, 11);
+    }
+
+    @Test
+    public void testFindByRegionsAndComplexFiltersExcludeSingleRoot() {
+        List<Region> regions = new ArrayList<>();
+        regions.add(new Region("11", 183000, 183300));
+
+        List<String> exclude = new ArrayList<>();
+        exclude.add("files");
+
+        List<VariantEntity> variantEntityList =
+                variantEntityRepository.findByRegionsAndComplexFilters(regions, null, null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null, exclude,
+                                                                       new PageRequest(0, 100000000));
+
+        for (VariantEntity currVariantEntity : variantEntityList) {
+            assertTrue(currVariantEntity.getSourceEntries().isEmpty());
+        }
+
+        exclude = new ArrayList<>();
+
+        variantEntityList =
+                variantEntityRepository.findByRegionsAndComplexFilters(regions, null, null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null,
+                                                                       VariantEntityRepository.RelationalOperator.NONE,
+                                                                       null, exclude,
+                                                                       new PageRequest(0, 100000000));
+
+        for (VariantEntity currVariantEntity : variantEntityList) {
+            assertFalse(currVariantEntity.getSourceEntries().isEmpty());
+        }
+    }
+
+    @Test
+    public void testFindByRegionsAndComplexFiltersExcludeMultiple() {
```

Comment:
One of the test cases (no exclusions) will be redundant if the previous one is split in 2. The asserts can be grouped under that single method.

Applies also to `testFindByRegionsAndComplexFiltersExcludeNested`.